src/openzaak/components/documenten/api/viewsets.py

```diff
@@ -170,14 +162,6 @@ class EnkelvoudigInformatieObjectViewSet(NotificationViewSetMixin,
 
     @transaction.atomic
     def perform_destroy(self, instance):
-        if instance.canonical.objectinformatieobject_set.exists():
```

Comment:
This should also be back in if we return OIO.

Reply:
I think it's checking for:
* `instance.canonical.besluitinformatieobject_set.exists()`
* `instance.canonical.zaakinformatieobject_set.exists()`

Reply:
oh,  I didn't think of them. The validation check and related test cases are added